R/PipeOpTextVectorizer.R

```diff
@@ -0,0 +1,251 @@
+        ParamDbl$new("min_termfreq", lower = 0, upper = Inf, default = NULL,
+          tags = c("train", "predict", "dfm_trim"), special_vals = list(NULL)),
+        ParamDbl$new("max_termfreq", lower = 0, upper = Inf, default = NULL,
+          tags = c("train", "predict", "dfm_trim"), special_vals = list(NULL)),
+
+        ParamFct$new("scheme", default = "unary", tags = c("train", "predict", "docfreq"),
+          levels = c("count", "inverse", "inversemax", "inverseprob", "unary")),
+        ParamFct$new("scheme_tf", default = "count", tags = c("train", "predict"),
+          levels = c("count", "prop", "propmax", "logcount", "boolean", "augmented", "logave")),
+        ParamDbl$new("smoothing", lower = 0, upper = Inf, default = 0, tags = c("train", "predict", "docfreq")),
+        ParamDbl$new("k", lower = 0, upper = Inf, default = 0, tags = c("train", "predict", "docfreq", "dfm_weight")),       
+        ParamDbl$new("threshold", lower = 0, upper = Inf, default = 0, tags = c("train", "predict", "docfreq", "dfm_weight")),
+        ParamDbl$new("base", lower = 0, upper = Inf, default = 10, tags = c("train", "predict", "docfreq"))
+      ))
+      ps$add_dep("base", "scheme", CondAnyOf$new(c("inverse", "inversemax", "inverseprob")))
+      ps$add_dep("smoothing", "scheme", CondAnyOf$new(c("inverse", "inversemax", "inverseprob")))
+      ps$add_dep("k", "scheme", CondAnyOf$new(c("inverse", "inversemax", "inverseprob")))
+      ps$add_dep("base", "scheme", CondAnyOf$new(c("inverse", "inversemax", "inverseprob")))
+      ps$add_dep("threshold", "scheme", CondAnyOf$new(c("inverse", "inversemax", "inverseprob")))
+      ps$values = list("remove_stopwords" = TRUE, language = "smart", n = 1, scheme = "unary", scheme_tf = "count")
+      super$initialize(id = id, param_set = ps, param_vals = param_vals, packages = c("quanteda", "stopwords"))
+    },
+
+    select_cols = function(task) {
+      task$feature_types[get("type") == "character", get("id")]
+    },
+
+    train_dt = function(dt, levels, target) {
+      tdm = private$transform_bow(dt)                     # transform to BOW, return doc-freq matrix
+      self$state = list(tdm = quanteda::dfm_subset(tdm, FALSE))     # empty tdm so we have vocab of training data
+      # tf-idf
+      if (self$param_set$get_values()$scheme != "unary") {
+          self$state$docfreq = invoke(quanteda::docfreq, .args = c(tdm, self$param_set$get_values(tags = "docfreq")))
+          tdm = private$transform_tfidf(tdm)
+      }
+      quanteda::convert(tdm, "matrix")
+    },
+    predict_dt = function(dt, levels, target) {
+      if (nrow(dt)) {
+        tdm = private$transform_bow(dt)
+        tdm = rbind(tdm, self$state$tdm) # make sure all columns occur
+        # tf-idf
+        if (self$param_set$get_values()$scheme != "unary") tdm = private$transform_tfidf(tdm)
+        tdm = tdm[, colnames(self$state$tdm)]   # Ensure only train-time features are pased on
+      } else {
+        tdm = self$state$tdm
```

Comment:
if we really want 100% test coverage, we should cover this (although I do not think we currently aim at 100% for most `PipeOp`)